src/Turing.jl

```diff
@@ -122,13 +124,13 @@ export  @model,                 # modelling
         logjoint,
         LogDensityFunction,
 
-        constrained_space,            # optimisation interface
-        MAP,
+        MAP,            # optimisation interface
         MLE,
-        get_parameter_bounds,
-        optim_objective,
-        optim_function,
-        optim_problem
+        OptimLogDensity,
+        ModeResult,
+        estimate_mode,
+        maximum_a_posteriori,
+        maximum_likelihood
```

Comment:
`MAP`, `MLE`, and `estimate_mode` no longer needs to be exported, right? `maximum_a_posteriori` and `maximum_likelihood` has us covered.

Reply:
They don't need to, yes. I could imagine someone preferring them as an interface, but happy to minimise the exported surface if you prefer.

Reply:
Agree but I think those are sufficient "power users" where they can just import them or do the qualifier anyways :+1:

Reply:
I left `MLE` and `MAP` since the Optim.jl interface needs for them to be exported. Can remove them later once we remove that extension, although will have to think about breakage and deprecation. Otherwise now down to just `maximum_likelihood` and `maximum_a_posteriori`.